src/main/java/com/exasol/adapter/metadata/converter/SchemaMetadataJsonConverter.java

```diff
@@ -0,0 +1,190 @@
+package com.exasol.adapter.metadata.converter;
+
+import java.util.Collections;
+
+import javax.json.*;
+
+import com.exasol.adapter.metadata.*;
+import com.exasol.adapter.metadata.DataType.*;
+
+/**
+ * This class converts the schema metadata to its JSON representation.
+ */
+public final class SchemaMetadataJsonConverter {
+    private static final SchemaMetadataJsonConverter instance = new SchemaMetadataJsonConverter();
+    private static final String ADAPTER_NOTES_KEY = "adapterNotes";
+    private static final String TABLES_KEY = "tables";
+    private static final String TABLE_TYPE_KEY = "type";
+    private static final String TABLE_NAME_KEY = "name";
+    private static final String DATA_TYPE_KEY = "dataType";
+    private static final String NULLABLE_KEY = "isNullable";
+    private static final String COMMENT_KEY = "comment";
+    private static final String IDENTITY_KEY = "isIdentity";
+    private static final String DEFAULT_KEY = "default";
+    private static final String DATA_TYPEKEY = "type";
+    private final JsonBuilderFactory factory = Json.createBuilderFactory(Collections.emptyMap());
+
+    /**
+     * Get the singleton instance of the {@link SchemaMetadataJsonConverter}
+     *
+     * @return converter instance
+     */
+    public static SchemaMetadataJsonConverter getInstance() {
+        return instance;
+    }
+
+    private SchemaMetadataJsonConverter() {
+    }
+
+    /**
+     * Convert the given schema metadata to its JSON representation
+     *
+     * @param schemaMetadata schemaMetadata to be converted into JSON
+     * @return JSON representation
+     */
+    public JsonObject convert(final SchemaMetadata schemaMetadata) {
+        final JsonObjectBuilder root = this.factory.createObjectBuilder();
```

Comment:
Maybe it should be moved closer to the first usage?

Reply:
Done.